src/posix/tty.rs

```diff
@@ -644,3 +646,29 @@ mod tests {
         assert_eq!(settings.flow_control(), Some(::FlowNone));
     }
 }
+
+pub fn list_ports() -> Vec<PortInfo> {
+    let mut vec = Vec::new();
+    if let Ok(context) = libudev::Context::new() {
+        let mut enumerator = match libudev::Enumerator::new(&context) {
+            Err(_) => return vec,
+            Ok(k) => k
+        };
+        if enumerator.match_subsystem("tty").is_err() {
+            return vec;
+        }
+        match enumerator.scan_devices() {
+            Ok(d) => {
+                for device in d {
+                    if let Some(_) = device.parent() {
+                        if let Some(path) = device.syspath().to_str() {
```

Comment:
Seems like `devnode` (`/dev/ttyUSB0`) would be more useful to the user than `syspath` (`/sys/devices/pci0000:00/0000:00:14.0/usb1/1-9/1-9:1.0/ttyUSB0/tty/ttyUSB0`).